### ISSUE TITLE
Saved location pinning/reordering

### ISSUE THREAD

**Issue by SynapseScribe** [davidtakac/prognoza#29]
ability to pin / unpin favorite locations

**Comment by davidtakac** [comment]
Hello. In this case, would pinned locations be displayed at the top of the list?

**Comment by SynapseScribe** [comment]
yes this is how i see it in the current layout. an icon should also be shown to indicate when it is pinned

**Comment by SynapseScribe** [comment]
maybe a nice idea would be to also allow reordering them and also clear other locations (by swiping it left). for the pinned ones the swipe to remove should be disabled, requiring to unpin to be able to remove. by swiping right you would pin/unpin a location

**Comment by davidtakac** [comment]
Clearing non-selected locations is possible since 3.2.0, you just have to long-click and a dialog pops up. 

I understand the pin and reorder ideas now, thank you. Do you have a lot of saved locations and require a mechanism to organize them? I'm just trying to understand your use case.

**Comment by SynapseScribe** [comment]
yes, i watch multiple locations. indeed I see now that i am able to delete locations on latest version by long press.
 for now, just a pin to top should do it, without extra stuff i suppose.

later on:
- i would add some dots on left that make the locations draggable to order them
- also one nice idea would be to show weather for each location, but i suppose that implies more api calls. so maybe you could limit by making api calls only for favourites and show the temperature only for these.

 I don't know how this api works, but it s nice that it doesn't require an api key like openweather does which enforced all apps and the developers to request users to create one

**Comment by davidtakac** [comment]
Maybe reordering could replace pinning altogether? It seems to be a more powerful mechanic. Pinning just floats the items to the top, but reordering allows you to do that yourself and whatever else you wish. 

> show weather for each location

I thought about that a while ago and it could be done. I could just fetch offline data for the location if it exists. 

> I don't know how this api works, but it s nice that it doesn't require an api key like openweather does which enforced all apps and the developers to request users to create one

It's a great API :) They don't require an API key, just a `UserAgent` header that identifies the developer, and that the developer respects their backend by distributing load properly.
